File: app/src/main/java/com/qimian233/ztool/hook/base/SystemHookModule.java
package com.qimian233.ztool.hook.base;

import io.github.libxposed.api.XposedModuleInterface;

/**
 * 系统框架 Hook 模块基类。
 * <p>
 * 继承此类后 IDE 会提示实现 {@link #handleSystemServerStarting}，无需关心
 * {@link #handleLoadPackage}。
 * </p>
 */
public abstract class SystemHookModule extends BaseHookModule {

    /**
     * 执行系统服务器启动时的 Hook 操作（<b>abstract</b>，便于 IDE 自动补全）。
     */
    @Override
    public abstract void handleSystemServerStarting(
            XposedModuleInterface.SystemServerStartingParam param) throws Throwable;
}

File: app/src/main/java/com/qimian233/ztool/hook/base/SystemHookModule.kt
package com.qimian233.ztool.hook.base

import io.github.libxposed.api.XposedModuleInterface

/**
 * Base class for system framework Hook modules (Kotlin).
 * <p>
 * After extending this class the IDE will prompt to implement
 * [handleSystemServerStarting]; [handleLoadPackage] is irrelevant here.
 * </p>
 */
abstract class SystemHookModule : BaseHookModule() {

    /**
     * Performs Hook operations on system server start (**abstract**, for IDE autocompletion).
     * <p>Carries [@Throws](Throwable::class) so Java subclasses can keep declaring {@code throws Throwable}.</p>
     */
    @Throws(Throwable::class)
    abstract override fun handleSystemServerStarting(
            param: XposedModuleInterface.SystemServerStartingParam)
}
